warehouse/index-stats/src/main/java/datawave/mapreduce/shardStats/StatsHyperLogMapper.java

```diff
@@ -0,0 +1,274 @@
+        log.info("normal precision(" + this.normalPrecision + ")");
+        this.sparsePrecision = conf.getInt(HYPERLOG_SPARSE_OPTION, HYPERLOG_SPARSE_DEFAULT_VALUE);
+        log.info("sparse precision(" + this.sparsePrecision + ")");
+        
+        this.visibility = new ColumnVisibility(conf.get(StatsJob.STATS_VISIBILITY));
+        
+        // initialize type registry
+        TypeRegistry.getInstance(conf);
+        
+        InputSplit split = context.getInputSplit();
+        log.info("Got a split of type " + split.getClass());
+        if (log.isInfoEnabled()) {
+            if (split instanceof TabletSplitSplit) {
+                TabletSplitSplit tabletSplit = (TabletSplitSplit) split;
+                log.info("Has " + tabletSplit.getLength() + " file range splits");
+                for (int i = 0; i < tabletSplit.getLength(); i++) {
+                    FileRangeSplit subSplit = (FileRangeSplit) tabletSplit.get(i);
+                    log.info("Got a file range spit of " + subSplit + " with a range of " + subSplit.getRange());
+                }
+            }
+        }
+        
+        if (log.isTraceEnabled()) {
+            log.trace("Completed setup(" + context + ")");
+        }
+    }
+    
+    @Override
+    protected void map(Key key, Value value, Context context) throws IOException, InterruptedException {
+        if (log.isTraceEnabled()) {
+            log.trace("map(" + key + ", " + value + ")");
+        }
+        
+        if (!setPartitionerContext) {
+            // This is a little bit fragile, but there is no other way to get the context
+            // on a partitioner, and we are only using this to set some counters
+            MultiTableRangePartitioner.setContext(context);
+            setPartitionerContext = true;
+        }
+        
+        // range should find all field index rows
+        String[] colf = StringUtils.split(key.getColumnFamily().toString(), NULL_CHAR);
+        if ("fi".equals(colf[0])) {
+            this.total++;
+            if (0 == this.total % this.logInputInterval) {
+                log.info("input row count(" + this.total + ")");
+            }
+            DatawaveKey dwKey = new DatawaveKey(key);
+            if (log.isTraceEnabled()) {
+                log.trace("mapper input(" + dwKey.toString() + ")");
+            }
+            
+            String fieldName = dwKey.getFieldName();
+            if (null == this.currentFieldName) {
+                this.currentFieldName = fieldName;
+            } else {
+                if (!this.currentFieldName.equals(fieldName)) {
+                    flushFieldValues(context);
+                    this.currentFieldName = fieldName;
+                }
+            }
+            
+            // add value to proper data type
+            String dataType = dwKey.getDataType();
+            DataTypeInfo typeInfo = this.dataTypeMapping.get(dataType);
```

Comment:
This is purely stylistic but in Java 8+, there is a `computeIfAbsent(key, key -> // compute initial value)` method on java.util.Map that can simplify the initialization of a Map entry.

Reply:
Yes, that is quite handy.